src/app/client/src/app/modules/shared/components/offline-card/offline-card.component.ts

```diff
@@ -70,7 +71,25 @@ export class OfflineCardComponent implements OnInit, OnChanges, OnDestroy {
     this.contentId = data.metaData.identifier;
     if (action === 'download') {
       data.showAddingToLibraryButton = true;
+      this.checkYoutubeContent(data, action);
+    } else {
+      this.clickEvent.emit({ 'action': action, 'data': data });
     }
+  }
+
+  checkYoutubeContent(data, action) {
```

Comment:
This complex logic can be simplified and moved to service since it is used in multiple components